HIPs/anoncreds.md

```diff
@@ -0,0 +1,171 @@
+
+## Status
+
+A new Hyperledger AnonCreds project would move very quickly to the “Graduated” status. The technology underlying AnonCreds dates back to 1998, with the core cryptographic algorithms published in 2004/2005. Hyperledger Indy is at least the second complete implementation (following [IDMixer](https://idemix.wordpress.com/), and possibly others). As we have recently found in the specification of AnonCreds and its use with other than Indy deployments, very little needs to be done to use AnonCreds with VDRs other than Indy. Most of the underlying infrastructure for the project is in place, including a community ready to move the technology forward.
+
+## Solution
+
+The Hyperledger AnonCreds project will consist of the following elements:
+
+* A Rust open source repository containing the implementation of AnonCreds currently in the Hyperledger Indy repository [indy-shared-rs](https://github.com/hyperledger/indy-shared-rs). The AnonCreds implementation will be separated from that repository, and minimally adjusted to provide a clear API for publishing and reading AnonCreds objects, and for producing current AnonCreds or W3C Verifiable Credential Data Model v1.1 verifiable credentials and presentations. Wrappers for the use of the C-callable Rust library will be provided for some languages, particularly those being used in Aries (Python, JavaScript, Java at least).
+  * Other implementations of AnonCreds may be added over time.
+* The JDF-based, [Community Specification v1.0](https://github.com/CommunitySpecification/1.0/blob/master/1._Community_Specification_License-v1.md)-licensed [AnonCreds Specification](https://anoncreds-wg.github.io/anoncreds-spec/), and its related [AnonCreds Methods Registry](https://anoncreds-wg.github.io/anoncreds-methods-registry/), along with the [Working Group](https://github.com/AnonCreds-WG) documenting the AnonCreds Specification will move to Hyperledger AnonCreds. The AnonCreds Methods registry contains a list of implementations of using AnonCreds with different distributed storage platforms, such as Hyperledger Indy, [cheqd.io](https://cheqd.io/), Hyperledger Fabric, Web Servers and so on.
+* Open source implementations of various AnonCreds Methods. Such methods may be hosted within Hyperledger AnonCreds or in separate open or closed source repositories. For example, AnonCreds Methods for using legacy Indy and `did:indy` would likely be sourced within the project, and other methods like HTTP (and/or `did:web`), Fabric and Besu are good candidates to be sourced within the project.
```

Comment:
Can you please clarify the statement where methods may be hosted in closed source repositories? The proposal’s core essence appears to be to provide VDR specific resolvers and registrars, and core common layer of methods that can be used in protocols like Aries.

Reply:
If you are familiar with the DID Method Registry, it is very much the same -- anyone can define and host their AnonCreds Method specification and implementation where ever they want.  Likewise, anyone that wants to support AnonCreds can provide a method for writing/reading AnonCreds objects. 

Closed Source example: If an implementation if for a proprietary system, they implementers might produce a proprietary plug-in for their Method.  

Open Source example: If an implementation is for an open source system (say a DLT), the maintainers might manage the AnonCreds Method implementation for their DLT along with the open source DLT.

Hyperledger AnonCreds example: AnonCreds methods that target either a ledger-less VDR or a ledger that is part of Hyperledger -- Indy, Fabric, Besu, etc.

Each way is fine.

Reply:
Thanks, got it.